src/biokbase/narrative/jobs/jobcomm.py

```diff
@@ -181,14 +187,21 @@ def __init__(self):
                 MESSAGE_TYPE["STOP_UPDATE"]: self._modify_job_updates,
             }
 
-    def _get_job_ids(self, req: JobRequest = None):
+    def _get_job_ids(self, req: JobRequest = None) -> List[str]:
+        """
+        Extract the job IDs from a job request object
+
+        :param req: the job request to take the IDs from, defaults to None
+        :type req: JobRequest, optional
+
+        :return: list of job IDs
+        :rtype: List[str]
+        """
+        if not req.has_job_ids():
+            raise JobRequestException(NO_INPUT_TYPE_ERR)
```

Comment:
Unneeded -- this was already handled by the code you deleted.

There isn't any need for distinction between `NO_INPUT_TYPE_ERR` and `ONE_INPUT_TYPE_ONLY_ERR` because the job request is sent back with the error and the frontend can't do anything helpful with the error message other than output it to the console.

Reply:
Okay, `ONE_INPUT_TYPE_ONLY_ERR` sounded to me like there were too many input types